blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -137,10 +138,13 @@ class RunDetails extends Component {
     afterClose = () => {
         const { router, params } = this.context;
         if (this.opener) {
-            router.goBack();
+            // step back in the history to the item that's prior to initial load of RunDetails
+            const offsetToInitialRoute = this.initialHistoryLength - history.length;
+            router.go(offsetToInitialRoute - 1);
```

Comment:
why not keep popping off history elements until we're away from the 'modal' ? damn modal...